idaes/models/unit_models/heat_exchanger_1D.py

```diff
@@ -287,198 +284,244 @@ def build(self):
+            has_pressure_change=self.config.cold_side.has_pressure_change,
         )
 
-        self.tube.apply_transformation()
+        self.cold_side.apply_transformation()
 
-        # Add Ports for shell side
-        self.add_inlet_port(name="shell_inlet", block=self.shell)
-        self.add_outlet_port(name="shell_outlet", block=self.shell)
+        # Add Ports for hot side
+        self.add_inlet_port(name="hot_side_inlet", block=self.hot_side)
+        self.add_outlet_port(name="hot_side_outlet", block=self.hot_side)
 
-        # Add Ports for tube side
-        self.add_inlet_port(name="tube_inlet", block=self.tube)
-        self.add_outlet_port(name="tube_outlet", block=self.tube)
+        # Add Ports for cold side
+        self.add_inlet_port(name="cold_side_inlet", block=self.cold_side)
+        self.add_outlet_port(name="cold_side_outlet", block=self.cold_side)
 
-        # Add reference to control volume geometry
-        add_object_reference(self, "shell_area", self.shell.area)
-        add_object_reference(self, "shell_length", self.shell.length)
-        add_object_reference(self, "tube_area", self.tube.area)
-        add_object_reference(self, "tube_length", self.tube.length)
+        # Add references to the user provided aliases if applicable
+        add_hx_references(self)
 
+        self._make_geometry()
         self._make_performance()
 
+        hot_side_units = (
+            self.config.hot_side.property_package.get_metadata().get_derived_units
+        )
+        q_units = hot_side_units("power") / hot_side_units("length")
+
+        @self.Constraint(
+            self.flowsheet().time,
+            self.hot_side.length_domain,
+            doc="Heat conservation equality",
+        )
+        def heat_conservation(self, t, x):
+            return pyunits.convert(self.cold_side.heat[t, x], to_units=q_units) == -(
+                self.hot_side.heat[t, x]
+            )
+
+    def _process_config(self):
+        hx_process_config(self)
+
+    def _make_geometry(self):
+        hot_side_units = (
+            self.config.hot_side.property_package.get_metadata().get_derived_units
+        )
+
+        self.area = Var(
+            initialize=1, units=hot_side_units("area"), doc="Heat transfer area"
+        )
+
+        # Add reference to control volume geometry
+        add_object_reference(self, "length", self.hot_side.length)
+
+        # Equate hot and cold side lengths
+        @self.Constraint(
+            self.flowsheet().time, doc="Equating hot and cold side lengths"
+        )
+        def length_equality(self, t):
+            return (
+                pyunits.convert(
+                    self.cold_side.length, to_units=hot_side_units("length")
+                )
+                == self.hot_side.length
+            )
+
+        # Also need to set cross-sectional areas for control volumes
+        # For now, fix ot a nominal value of 1
+        self.cold_side.area.fix(1)
+        self.hot_side.area.fix(1)
```

Comment:
For most cases the user does not need to know about these - they are only important in cases where velocities are required (and maybe for some dynamic simulations). To avoid unexpected degrees of freedom in these cases, I decided to fix these, and the documentation will tell users about this.

Note that these are not indexed variables, so there are no issues with DAE transformations,

Reply:
> To avoid unexpected degrees of freedom in these cases, I decided to fix these

So there are equations in the `ControlVolume1D` using these?

> Note that these are not indexed variables, so there are no issues with DAE transformations,

Yeah, I had something about that at first, but deleted it when I realized they weren't indexed.

Reply:
I believe for the most part they do not appear in any equations for steady-state models (and thus will never be seen by the solver even if it is not fixed), however they do appear in the holdup terms for dynamic models. Even for dynamic cases, unless the area changes over time (which doesn't happen in most simple models), the actual value of area doesn't matter (as long as it is fixed).

However, the variable is always created - in an ideal world we would probably have some set of checks in place for this, but we don't.

Reply:
Why wouldn't the area matter for dynamic models? The larger the area, the more fluid that's sitting in the pipes and needs to be flushed out.

Now, this base class wouldn't be *good* for dynamics, because it doesn't take heat exchanger metal mass into account. For liquids, that's a large part of the heat capacity, and, for gases, it's practically the *entire* heat capacity.

As a related comment, I'd like an option to avoid creating control volume holdup without setting `"has_holdup": False` for subclasses so they can exclude those small terms for gas phase streams (and avoid the extremely fast dynamics that happen).

Reply:
True, the value does matter for dynamics (which is why area always exists) - I am getting lost in other details.

`has_holdup` defaults to False (expect for dynamic models where it is required), so you don't actually need to pass an argument if you never want holdup to appear. For overall dynamic flowsheets, remember that it is valid to put steady-state models in dynamic systems to represent cases with very fast dynamics (specifically to support cases like you mentioned).

Reply:
I was thinking about passing `False` for the control volumes but still having it `True` for the overall unit (and thus allowing for dynamics). Fortunately, that seems to be possible with the current setup.

Reply:
Yes - that was the intent behind that decision.

Back on the original topic, I think you have convinced me to leave these unfixed - for steady state models it won;t matter, and for dynamic models it is better the users sees unexpected degrees of freedom so they know to set these values.